Source/Core/VideoCommon/Widescreen.cpp

```diff
@@ -31,80 +54,101 @@ WidescreenManager::WidescreenManager()
     return;
 
-  Update();
-
-  // If widescreen hack isn't active and aspect_mode (user setting)
-  // is set to a forced aspect ratio, don't use heuristic.
-  if (!g_ActiveConfig.bWidescreenHack && (g_ActiveConfig.aspect_mode == AspectMode::ForceStandard ||
-                                          g_ActiveConfig.aspect_mode == AspectMode::ForceWide))
-    return;
-
-  // Modify the threshold based on which aspect ratio we're already using:
-  // If the game's in 4:3, it probably won't switch to anamorphic, and vice-versa.
-  const u32 transition_threshold = g_ActiveConfig.widescreen_heuristic_transition_threshold;
-
-  const auto looks_normal = [transition_threshold](auto& counts) {
-    return counts.normal_vertex_count > counts.anamorphic_vertex_count * transition_threshold;
-  };
-  const auto looks_anamorphic = [transition_threshold](auto& counts) {
-    return counts.anamorphic_vertex_count > counts.normal_vertex_count * transition_threshold;
-  };
-
-  const auto& persp = flush_statistics.perspective;
-  const auto& ortho = flush_statistics.orthographic;
+  std::optional<bool> is_game_widescreen = GetWidescreenOverride();
 
-  const auto ortho_looks_anamorphic = looks_anamorphic(ortho);
-
-  if (looks_anamorphic(persp) || ortho_looks_anamorphic)
-  {
-    // If either perspective or orthographic projections look anamorphic, it's a safe bet.
-    m_is_game_widescreen = true;
-  }
-  else if (looks_normal(persp) || (m_was_orthographically_anamorphic && looks_normal(ortho)))
+  // If widescreen hack isn't active and aspect_mode (UI) is 4:3 or 16:9 don't use heuristic.
+  if (g_ActiveConfig.bWidescreenHack || (g_ActiveConfig.aspect_mode != AspectMode::ForceStandard &&
+                                         g_ActiveConfig.aspect_mode != AspectMode::ForceWide))
   {
-    // Many widescreen games (or AR/GeckoCodes) use anamorphic perspective projections
-    // with NON-anamorphic orthographic projections.
-    // This can cause incorrect changes to 4:3 when perspective projections are temporarily not
-    // shown. e.g. Animal Crossing's inventory menu.
-    // Unless we were in a situation which was orthographically anamorphic
-    // we won't consider orthographic data for changes from 16:9 to 4:3.
-    m_is_game_widescreen = false;
+    // Modify the threshold based on which aspect ratio we're already using:
+    // If the game's in 4:3, it probably won't switch to anamorphic, and vice-versa.
+    const u32 transition_threshold = g_ActiveConfig.widescreen_heuristic_transition_threshold;
+
+    const auto looks_normal = [transition_threshold](auto& counts) {
+      return counts.normal_vertex_count > counts.anamorphic_vertex_count * transition_threshold;
+    };
+    const auto looks_anamorphic = [transition_threshold](auto& counts) {
+      return counts.anamorphic_vertex_count > counts.normal_vertex_count * transition_threshold;
+    };
+
+    const auto flush_statistics = g_vertex_manager->ResetFlushAspectRatioCount();
```

Comment:
This should be moved back up to the start of the function. `ResetFlushAspectRatioCount()` not only returns the counts but also resets them to zero, so if you don't do that you end up accumulating counts over multiple frames for every frame where this line isn't reached.

Reply:
good catch. fixed.